src/main/java/com/gluonhq/substrate/model/InternalProjectConfiguration.java

```diff
@@ -474,25 +474,38 @@ public boolean usesJDK11() {
      * Check if the GraalVM provided by the configuration is supported
      * @throws IOException if the GraalVM version is older than the minimum supported version
      */
-    public void checkGraalVMVersion() throws IOException {
-        Version graalVersion = getGraalVersion();
-        if (graalVersion.compareTo(new Version(Constants.GRAALVM_MIN_VERSION)) < 0) {
-            throw new IOException("Current GraalVM version (" + graalVersion + ") not supported.\n" +
-                    "Please upgrade to " + Constants.GRAALVM_MIN_VERSION + " or higher");
+    public void checkGraalVMVersion(Version javaVersion) throws IOException {
+        if (isOldGraalVMVersioningScheme(javaVersion)) {
+            Version graalVersion = getGraalVersion();
+            if (graalVersion.compareTo(new Version(Constants.GRAALVM_MIN_VERSION)) < 0) {
+                throw new IOException("Current GraalVM version (" + graalVersion + ") not supported.\n" +
+                        "Please upgrade to " + Constants.GRAALVM_MIN_VERSION + " or higher");
+            }
+        }
```

Comment:
This means that if there is a GraalVM build that uses the new GraalVM versioning system, there is no check at all if this build is older than the GraalVM_MIN_VERSION.
Of course, currently the min version is "22.1" (old system), so _any_ build with the new versioning system is newer, and whenever we increase that min value in the future, we should use the new versioning system and do a proper check at that point. 
For now, maybe adding something like: 
```
} else {
    // TODO: check the GraalVM version whenever Constants.GRAALVM_MIN_VERSION is bumped and uses the new versioning system
}
```
will do?